zerver/openapi/zulip.yaml

```diff
@@ -15647,3 +14751,404 @@ components:
+      in: query
+      description: |
+        Enable audible desktop notifications for private messages and
+        @-mentions.
+      schema:
+        type: boolean
+      example: true
+
+    EnableStreamAudibleNotifications:
+      name: enable_stream_audible_notifications
+      in: query
+      description: |
+        Enable audible desktop notifications for stream messages.
+      schema:
+        type: boolean
+      example: true
+
+    EnableStreamDesktopNotifications:
+      name: enable_stream_desktop_notifications
+      in: query
+      description: |
+        Enable visual desktop notifications for stream messages.
+      schema:
+        type: boolean
+      example: true
+
+    EnableStreamEmailNotifications:
+      name: enable_stream_email_notifications
+      in: query
+      description: |
+        Enable email notifications for stream messages.
+      schema:
+        type: boolean
+      example: true
+
+    EnableStreamPushNotifications:
+      name: enable_stream_push_notifications
+      in: query
+      description: |
+        Enable mobile notifications for stream messages.
+      schema:
+        type: boolean
+      example: true
+
+    EnterSends:
+      name: enter_sends
+      in: query
+      description: |
+        Whether pressing Enter in the compose box sends a message
+        (or saves a message edit).
+      schema:
+        type: boolean
+      example: true
+
+    EscapeNavigatesToDefaultView:
+      name: escape_navigates_to_default_view
+      in: query
+      description: |
+        Whether the escape key navigates to the
+        [configured default view](/help/configure-default-view).
+
+        **Changes**: New in Zulip 5.0 (feature level 107).
+      schema:
+        type: boolean
+      example: true
+
+    FluidLayoutWidth:
+      name: fluid_layout_width
+      in: query
+      description: |
+        Whether to use the [maximum available screen width](/help/enable-full-width-display)
+        for the web app's center panel (message feed, recent topics) on wide screens.
+      schema:
+        type: boolean
+      example: true
+
+    HighContrastMode:
+      name: high_contrast_mode
+      in: query
+      description: |
+        This setting is reserved for use to control variations in Zulip's design
+        to help visually impaired users.
+      schema:
+        type: boolean
+      example: true
+
+    LeftSideUserlist:
+      name: left_side_userlist
+      in: query
+      description: |
+        Whether the users list on left sidebar in narrow windows.
+
+        This feature is not heavily used and is likely to be reworked.
+      schema:
+        type: boolean
+      example: true
+
+    MessageContentInEmailNotifications:
+      name: message_content_in_email_notifications
+      in: query
+      description: |
+        Include the message's content in email notifications for new messages.
+      schema:
+        type: boolean
+      example: true
+
+    NotificationSound:
```

Comment:
I wonder if it'd be better to name these parameters e.g. `notification_sound` so that one doesn't need to have two formats of name for the same concept?  

I think the main potential downside is some consequence in what classes/objects get generated by openapi-generator, but that pattern for individual fields like these would make this a lot more greppable.

Reply:
I think it's a style preference on our part for naming conventions for component parameters. I believe the components are just references within the OpenAPI documentation to reduce duplication.

For example, I noticed that all but one of the component schema in the document are camelcase except for `profile_data`. Perhaps that happened for a similar thought process around making the content more greppable?

I do think there should be consistency in the document around the naming conventions though, so perhaps a conversation for chat/zulip?